apps/web/src/components/forms/status-page/form.tsx

```diff
@@ -115,35 +115,41 @@ export function StatusPageForm({
   }, [watchTitle, form, defaultValues?.title]);
 
   const onSubmit = async ({ ...props }: InsertPage) => {
-    startTransition(async () => {
-      try {
-        if (defaultValues) {
-          await api.page.update.mutate(props);
-        } else {
-          const page = await api.page.create.mutate(props);
-          const id = page?.id || null;
-          router.replace(`?${updateSearchParams({ id })}`); // to stay on same page and enable 'Advanced' tab
-        }
+    const isUnique = await checkUniqueSlug();
+    if (!isUnique) {
+      // the user will already have the "error" message - we include a toast as well
+      toastAction("unique-slug");
+    } else {
```

Comment:
Thanks @Ipriyankrajai! One thing: can we move the async call within the `startTransition` so that the pending state gets fired immediately?

Reply:
Sure! I was thinking that only